clib/src/main/scala/scala/scalanative/libc/linux/epoll.scala

```diff
@@ -0,0 +1,95 @@
+package scala.scalanative
+package libc
+package linux
+
+/*
+ * https://man7.org/linux/man-pages/man2/epoll_wait.2.html
+ * https://man7.org/linux/man-pages/man2/epoll_create1.2.html
+ */
+
+import unsafe._
+import unsigned._
+
+@extern
+@define("__SCALANATIVE_C_EPOLL")
+private[scalanative] trait epoll {
+
+  @name("scalanative_epoll_cloexec")
+  def EPOLL_CLOEXEC: CInt = extern
+
+  @name("scalanative_epoll_ctl_add")
+  def EPOLL_CTL_ADD: CInt = extern
+  @name("scalanative_epoll_ctl_mod")
+  def EPOLL_CTL_MOD: CInt = extern
+  @name("scalanative_epoll_ctl_del")
+  def EPOLL_CTL_DEL: CInt = extern
+
+  @name("scalanative_epollin")
+  def EPOLLIN: CInt = extern
+  @name("scalanative_epollout")
+  def EPOLLOUT: CInt = extern
+  @name("scalanative_epollrdhup")
+  def EPOLLRDHUP: CInt = extern
+  @name("scalanative_epollpri")
+  def EPOLLPRI: CInt = extern
+  @name("scalanative_epollerr")
+  def EPOLLERR: CInt = extern
+  @name("scalanative_epollhup")
+  def EPOLLHUP: CInt = extern
+
+  @name("scalanative_epollet")
+  def EPOLLET: CInt = extern
+  @name("scalanative_epolloneshot")
+  def EPOLLONESHOT: CInt = extern
+
+  type epoll_data_t = CStruct1[CLongLong] // 64-bit
+  type epoll_event = CStruct2[UInt, epoll_data_t]
+
+  def epoll_create(size: CInt): CInt = extern
+  def epoll_create1(flags: CInt): CInt = extern
+
+  def epoll_ctl(
+      epfd: CInt,
+      op: CInt,
+      fd: CInt,
+      event: Ptr[epoll_event]
+  ): CInt = extern
+
+  @blocking
```

Comment:
You've mentioned that you don't know why it's needed - `@blocking` annotations tells the GC to not wait for thread that is executing a possibly blocking call, this ones are more expensive, since these require to save some state before calling the actual foreign function, but allows us to not use signals to interupt code which would require us to guard almost all usages. If there is more functions that can block for undefined (or long) ammont of time these should be marked with this annotation as well